documentation/tutorials/deploy-kernel.rst

```diff
@@ -0,0 +1,179 @@
+  talking about derivative distros). For example, Ubuntu and Mint are derivated
+  from Debian, meaning that kw supports those distros.
+* Right now, kw only supports GRUB2 bootloader.
+
+Notice that these limitations can change in the future; consider to help us to
+improve this in the future.
+
+Tutorial Assumptions
+--------------------
+
+Now that you know kw limitations, we also assume the following tasks are
+completed:
+
+* You already compiled your kernel 
+* You have a functional `.config` file
+* For remote deploy, you have root access to the target machine.
+* All commands in this tutorial must be executed inside a kernel tree.
+
+.. note::
+    We artificially defined those assumptions to make this tutorial short.
+    Still, kw deploy has many features, do not hesitate to look at the man page
+    more for more details.
+
+Remote Machine Deploy
+---------------------
+
+Before trying to deploy your new kernel, let's first update the
+`kworkflow.config` by making sure that you set the following options are
+correct::
+
+   ssh_remote=<IP or NAME>
+   ssh_port=22
+
+Replace the `ssh_remote` by the IP or the destination name and the ssh port
+(the default port is 22, and you probably don't need to update this parameter).
+
+.. note::
+   If you don't know anything about `kworkflow.config`, take a look at
+   :ref:`kw configuration<setup-tutorial>`.
+
+Now, make sure that you can use kw to login into your target machine::
+
+  kw ssh
+
+If you cannot ssh to the machine, stop here and figure out why you cannot ssh
+to the target machine; after that, get back to the part.
+
+.. note::
+   If it is your first time trying to ssh to the remote machine, try the
+   following command inside it:
+   `sed -i '/#PermitRootLogin without-password/c\PermitRootLogin yes' /etc/ssh/sshd_config && systemctl restart sshd`
+
+If your ssh key has a password, we recommend you to use the below commands
+before you try to deploy your new kernel::
+
+  ssh-agent bash
+  ssh-add ~/.ssh/id_rsa
+
+Since this is the first time that you deploy a new kernel using kw, let's start
+with a command that setup your target machine::
+
+  kw deploy --setup
+
+.. note::
+   All the above configuration represents a one-time setup.
+
+Under the hood, the above command will:
+
+- Make sure that the ssh config is correct.
+- Install required packages in the target machine.
+- Install a small part of kw in the remote.
+
+After the above command completes, you can deploy your change by using the
+command::
+
+ kw deploy # kw d
+
+That's it, kernel installed! You just need to reboot your target system and
+select your new kernel in the GRUB menu.
+
+Local Machine Deploy
+--------------------
+
+In this scenario, a target kernel is your host machine. For example, suppose
+that you want to install the latest stable kernel from the Torvalds tree in
+your laptop; in this case, kw deploy --local is what you are looking for.
```

Comment:
Maybe it is a `newbie` thought, but it would be good to advise the risk of doing it and even breaking the personal/local machine. There is this risk for remote deployment in a bad ssh configuration, but it is less risky because you mentioned the SSH connection first (that can prevent the issue for anxious people).